src/condor_startd.V6/command.cpp

```diff
@@ -1145,73 +1136,86 @@ request_claim( Resource* rip, Claim *claim, char* id, Stream* stream )
 		//   sending to WorkingCM?
 		rip->dprintf(D_FULLDEBUG, "Refusing claim of pslot with consumption policy\n");
 		refuse(stream);
-		ABORT;
+		goto abort;
 	}
 
 	// The client can claim at most one new dslot.
 	// This will change in the future.
-	if (num_dslots < 0 || num_dslots > 1) {
+	if (num_dslots < 0 || (num_dslots > 1 && rip->r_has_cp)) {
```

Comment:
I believe the check for r_has_cp is redundant here, since earlier you force num_dslots to 1 if r_has_cp is true.

Reply:
true, but this is clearer.